### PR TITLE
RHEL 8.9 nullptr dereference

### PR DESCRIPTION
**Pull request by AlexScherba16**
- changes in block_device in kernel
- support new linux kernel function declarations
- support for bd_has_submit_bio for block device

### CODE REVIEW

**Review by st-matskevich — APPROVED** [review]

**Comment by Swistusmen** [comment]
Is it all copied from https://github.com/datto/dattobd/compare/main...Swistusmen:dattobd:supporting-ubuntu24 and https://github.com/datto/dattobd/pull/367 + changes made here: https://github.com/datto/dattobd/pull/377/commits/e9e8df1904301842cc6ab659068dba31ecc44fec?

**Comment by AlexScherba16** [comment]
@Swistusmen without changes in tracer.

File: src/blkdev.c [hunk 1]
```diff
@@ -85,15 +80,108 @@ struct block_device *blkdev_get_by_path(const char *pathname, fmode_t mode,
...
+ * dattobd_drop_super() - Releases the superblock of the file system.
+ * This function performs the appropriate action based on the available
+ * kernel functions to release or drop the superblock.
+ *
+ * @sb: super block structure pointer to be released.
+ *
+ * Return:
+ * void.
+ */
+void dattobd_drop_super(struct super_block *sb)
```
Comment (Swistusmen) [inline src/blkdev.c:153]: Pls simplify to #if defined HAVE_GET_SUPER return drop_super(sb) #endif return;

**Review by Swistusmen — CHANGES_REQUESTED** [review]
Pls simplify dattobd_drop_super

**Comment by iamandrii** [comment]
Included to #382

**Pull request closed without merging**